src/condor_schedd.V6/qmgmt.cpp

```diff
@@ -9143,6 +9153,25 @@ bool BuildPrioRecArray(bool no_match_found /*default false*/) {
 
 	PrioRecArrayTimeslice.setFinishTimeNow();
 
+	// Now that we have rebuilt the PrioRec array, PrioRecMinCoolDownTime has been updated
+	// to the number of seconds until the next job leaves cool down.  Set a timer to
+	// call DirtyPrioRecArray() when that time is up. 
+	if ( DirtyPrioRecTid >= 0 ) {
+		daemonCore->Cancel_Timer(DirtyPrioRecTid);
+		DirtyPrioRecTid = -1;
+	}
+	if ( PrioRecMinCoolDownTime > 0 ) {
+		time_t seconds_until_cooldown = PrioRecMinCoolDownTime - time(nullptr);
+		// seconds_until_cooldown can be negative if building the PrioRec array
+		// took longer than the time to release the next cool down job.
+		// If this happens, set it to 0 so dirty the prio rec array again immediately.
+		if ( seconds_until_cooldown < 0 ) {
+			seconds_until_cooldown = 0;
+		}
+		DirtyPrioRecTid = daemonCore->Register_Timer(seconds_until_cooldown,
+				&DirtyPrioRecArray, "DirtyPrioRecArray");
+	}
+
```

Comment:
If min cooldown time is > next prio-rec rebuild, there is no need to  run a timer.

Reply:
Todd T called this a minor optimization. We will merge as is.